include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -31,6 +31,15 @@
 #    define __LIBSYCL_VERSION 0
 #endif
 
+#if __LIBSYCL_VERSION >= 50300
+#    define _ONEDPL_SYCL_SIZE size()
```

Comment:
I thing it is bad approach... especially `buf._ONEDPL_SYCL_SIZE`
 
I suggest using our own simple function-wrapper (in the SYCL backend namespace) which accepts a sycl buffer and returns size, f.e. `__par_backend_hetero::buffer_size(buf)`